docs/dev_env.rst

```diff
@@ -37,16 +37,21 @@ Getting the Source Code
 Create a fork
 ^^^^^^^^^^^^^
 To limit the number of branches created on the Apache Pinot repository, we recommend that you create a fork by clicking on the fork button `here <https://github.com/apache/incubator-pinot>`_.
+Read more about `fork workflow here <https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow>`_
 
 Clone the repository locally
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 .. code-block:: none
 
-  $mkdir workspace
-  $cd workspace
-  $git clone https://github.com/<username>/incubator-pinot.git
-  $cd incubator-pinot
+  $ mkdir workspace
```

Comment:
Can we move the section on how to create a fork into the contribution guidelines? We seem to say here that creating a branch is not recommended, yet that is what we write about in contribution guidelines. Maybe we should remove the text on creating branches, and just retain one on creating forks? Not sure what the limitations are in terms of branches created. Does Apache have a limit? Or, are we worried that people may not delete the branch after commit?

Reply:
A branch on the Apache repo is not recommended. We almost always need branches.

Reply:
On Apache repo, we would like to see branches that are used for releases (and not be cluttered with everything else that the community is working on)

Reply:
In that case, isn't it better to remove the other section on creating branches?